lib/Backends/OpenCL/OpenCL.cpp

```diff
@@ -1218,6 +1249,30 @@ void OpenCLFunction::execute() {
       continue;
     }
 
+    if (auto *TK = dyn_cast<TopKInst>(&I)) {
+      clFinish(commands_);
```

Comment:
Add a comment that you perform the computation on the host and to do that you copy the tensor from GPU memory to host memory, perform the operation and then copy the results back to GPU memory.